### PR TITLE
Add clang formatting checks for src/ files

### PR DESCRIPTION
**Pull request by willhoy**
**Describe your changes**
Adding a formatting action to preserve the clang formatting of the codebase

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -55,18 +55,30 @@ jobs:
...
     steps:
       - uses: actions/checkout@v3
       - name: Set up dependencies
         run: |
           sudo apt-get update
-          sudo apt-get install -qy clang-format git doxygen
-      - name: Lint sources
-        run: |
-          git clang-format
+          sudo apt-get install -qy git doxygen
```
Comment (willhoy) [inline .github/workflows/build.yaml:66]:
```suggestion
          sudo apt-get install -qy doxygen
```

Comment (willhoy) [reply .github/workflows/build.yaml:66]: I doubt this is necessary?

**Comment by hallfox** [comment]
I see that it hit a generated file. I think we should use the `exclude-regex` option in the clang-format checks to exclude mqbcfg_messages.(h|cpp)

**Comment by hallfox** [comment]
Going by other generated files in this repo I think we'll need to exclude .*_(messages|values|cpp03).(h|cpp)

**Pull request merged**